src/ground_script_1.c

```diff
@@ -0,0 +1,1863 @@
+                if (sub_809D8EC(action, curCmd.argShort)) {
+                    return 2;
+                }
+                break;
+            }
+            case 0xe4: {
+                sub_809D9B8(curCmd.argShort);
+                break;
+            }
+            case 0xe5: {
+                scriptData->branchDiscriminant = curCmd.argShort;
+                if ((s8)sub_809DA08(action, curCmd.argShort, curCmd.argByte)) {
+                    return 2;
+                }
+                break;
+            }
+            case 0xe6: {
+                scriptData->savedScript = scriptData->script;
+            } //fallthrough
+            case 0xe7: { // ???
+                scriptData->script.ptr = sub_80A242C(action, curCmd.argShort);
+                break;
+            }
+            case 0xe8: { // SAVE_AND_TRIGGER
+                scriptData->savedScript = scriptData->script;
+            } // fallthrough
+            case 0xe9: { // TRIGGER
+                scriptData->script.ptr2 = scriptData->script.ptr = gFunctionScriptTable[curCmd.argShort].script;
+                break;
+            }
+            case 0xea: { // SAVE_AND_RUN_STATION
+                scriptData->savedScript = scriptData->script;
+            } // fallthrough
+            case 0xeb: { // RUN_STATION
+                ScriptInfoSmall info;
+                u32 group, sector;
+                u32 tmp = gUnknown_2039A34;
+                {
+                    s32 tmp = curCmd.argShort < 0 ? scriptData->script.group : curCmd.argShort;
+                    group = tmp;
+                }
+                {
+                    s32 tmp = (s8)curCmd.argByte < 0 ? scriptData->script.sector : (s8)curCmd.argByte;
+                    sector = tmp;
+                }
+                GroundMap_GetStationScript(&info, tmp, group, sector);
+                scriptData->script.ptr = info.ptr;
+                scriptData->script.ptr2 = info.ptr;
+                scriptData->script.group = group;
+                scriptData->script.sector = sector;
+                break;
+            }
+            case 0xec: {
+                gUnknown_2039A34 = sub_80A4D7C((s16)sub_8001658(scriptData->unk50, curCmd.argShort));
+                sub_809D940();
+                GroundMap_ExecuteEnter(gUnknown_2039A34);
+                break;
+            }
+            case 0xed: { // DELETE_SAVED
+                scriptData->savedScript.ptr = NULL;
+                scriptData->savedScript.ptr2 = NULL;
+                break;
+            }
+            case 0xee: { // RESTORE_SAVED
+                if (scriptData->savedScript.ptr == NULL) {
+                    return 0; // huh?
+                }
+                scriptData->script = scriptData->savedScript;
+                scriptData->savedScript.ptr = NULL;
+                scriptData->savedScript.ptr2 = NULL;
+                break;
+            }
+            case 0xef: // RETURN_FALSE
```

Comment:
Is it worth it to create an enum for all these cases? I recall you have extensive notes on each case plus what's available online.

Reply:
Well, I don't have a complete description of all cases, so most enum variants would be `SCRIPT_CMD_6C` and such. And besides, I'm pretty sure they used raw integers here (see case ranges).
We can define macros for script commands when defining scripts in data/*.